### ISSUE TITLE
Game breaks and save file is corrupted after AI makes a Declaration of War.

### ISSUE THREAD

**Issue by Bloodrocution** [LoneGazebo/Community-Patch-DLL#5097]
_1. Mod version (i.e Date - 4/23):_
1-15 VP with EUI.

_2. Mod list (if using Vox Populi only, leave blank):_

![mod list](http://puu.sh/CzFny/f8207763b5.jpg)

_3. Error description:_ 

Game becomes "corrupted", for lack of a better description on a specific turn where many events take place: Carthage declares war on me and on Arabia, and Egypt and I both unlock our Reformation beliefs at the same time. 

When Carthage's leader screen pops up making her DoW speech, I can only get out of it with the "Very well..." dialogue option, and not the "You'll pay for this...". Ok, no big deal I guess, but immediately upon exiting this screen, some other weird things start going on: 1) a "Carthage has declared war" dialogue box shows up on the bottom of the screen, even before the AI turns have finished processing, and you have to click the "back" button to get out of it once they eventually do; then my cursor disappears and I'm unable to do anything until I press the escape key, which then brings up my reformation belief selection options. The cursor from this point on stays as the loading globe instead of the normal pointer. Weird graphical glitches/artifacts then begin to manifest if I continue playing out the turn, where different map tiles don't appear to load/render properly, but instead leave some wonky low-resolution textures scattered about randomly. The game doesn't crash, but I tried reloading the game after this bug occurs to try to remedy things and at this point it does crash.

_4. Steps to reproduce (optional):_

I'll attach the save file from right before the turn the bug takes places. I've tried playing it twice more to rule out it being some random fluke and the same thing happened. Just load the save and proceed to the next turn and that should reproduce the problem.

---------------------------
Supporting information:
Please note that you can attach .zip files by dragging-and-dropping them. If possible, zip up all supporting data and post that way.

1. Log files (always attach your Logs folder, located at My Documents/My Games/Sid Meier's Civilization 5. Make sure you have enabled logging before experiencing an error! Go here to find out how: http://forums.civfanatics.com/showthread.php?t=487482):

[Civ5logs.zip](https://github.com/LoneGazebo/Community-Patch-DLL/files/2776959/Civ5logs.zip)

2. Save game (always attach a save that was made a turn before the error; located at My Documents/My Games/Sid Meier's Civilization 5/ModdedSaves):

[Haile Selassie_0210 AD-0550.zip](https://github.com/LoneGazebo/Community-Patch-DLL/files/2776960/Haile.Selassie_0210.AD-0550.zip)

3. CvMiniDump.dmp file (attach if experiencing a game crash. Located at Program Files/Steam/steamapps/common/Sid Meier's Civilization V):

[CvMiniDump.zip](https://github.com/LoneGazebo/Community-Patch-DLL/files/2776961/CvMiniDump.zip)

4. Screenshots (optional):

**Comment by Bloodrocution** [comment]
Different game with slightly different mods enabled and I experienced the bug again! But it appears the only thing that triggers it is the AI's declaration of war (no religion belief selection this time around).

_1. Mod version (i.e Date - 4/23):_

1-29-1

_2. Mod list (if using Vox Populi only, leave blank):_

- (1) Community Patch	Version	89
- (2) Community Balance Overhaul	Version	13
- (3) City-State Diplomacy Mod for CBP	Version	27
- (4) C4DF - CBP	Version	11
- (5) More Luxuries - CBO Edition (5-14b)	Version	155
- (6a) Community Balance Overhaul - Compatibility Files (EUI)	Version	1
- (ui) Trade Opportunities for VP	Version	8
- CIV-Linked Great Generals	Version	11
- Enhanced Naval Warfare for Vox Populi	Version	1
- Enlightenment Era (Vox Populi)	Version	1
- Faster Aircraft Animations	Version	3
- Ingame Editor	Version	39
- Promotion Flags EUI	Version	1
- Really Advanced Setup	Version	15
- Reseed!	Version	11
- Tectonic Map Script	Version	4
- UI - Filters in Trade Route Overview (Vox Populi)	Version	1
- VP Promo Icons	Version	1
- Wonder Race	Version	7
- Unique City-States	Version	4

_3. Error description:_ 

See above. Same symptoms as before. AI makes a DOW; cannot select "You'll pay for this..." dialogue option"; "AI has declared war on you" dialogue window shows on the bottom of screen with a "back" button that you must click in order to close; cursor stays as the "loading" globe during player turn and map textures start to disappear; save file is corrupted and cannot be loaded.

_4. Steps to reproduce (optional):_

Load save. Proceed to next turn. I will attach both the save of the turn before the bug occurs as well as the corrupted save this time if that helps.

---------------------------
Supporting information:
Please note that you can attach .zip files by dragging-and-dropping them. If possible, zip up all supporting data and post that way.

1. Log files (always attach your Logs folder, located at My Documents/My Games/Sid Meier's Civilization 5. Make sure you have enabled logging before experiencing an error! Go here to find out how: http://forums.civfanatics.com/showthread.php?t=487482):

[Logs_Ramkhamhaeng.zip](https://github.com/LoneGazebo/Community-Patch-DLL/files/2839275/Logs_Ramkhamhaeng.zip)

2. Save game (always attach a save that was made a turn before the error; located at My Documents/My Games/Sid Meier's Civilization 5/ModdedSaves):

Corrupted save:
[Ramkhamhaeng_0143 AD-0660.zip](https://github.com/LoneGazebo/Community-Patch-DLL/files/2839276/Ramkhamhaeng_0143.AD-0660.zip)

Save of turn prior to bug occurring:
[QuickSave.zip](https://github.com/LoneGazebo/Community-Patch-DLL/files/2839277/QuickSave.zip)


3. CvMiniDump.dmp file (attach if experiencing a game crash. Located at Program Files/Steam/steamapps/common/Sid Meier's Civilization V):

4. Screenshots (optional):

**Comment by TechpriestEnginseer** [comment]
soo what if u use IGE and force the declaration of war earlier.

**Comment by Bloodrocution** [comment]
Yes, that does circumvent the bug from occurring. So something to do with the DoW leader screen is causing the bug maybe?

And while this method works as a stop-gap solution, I don't really see it as a "fix", so I'll keep the issue open. It robs a lot of the fun when you have to reload your game and force a DoW on the AI's behalf :( .

**Comment by Bloodrocution** [comment]
I started a third new game (same mod list as the last update) and had another occurrence of this bug, but somehow it didn't corrupt my save and cause the game to go completely haywire this time. What's different in this instance is that I'm currently at war with another faction, Japan (incidentally, a war which was also initiated by the AI, but without any hiccups during the DoW execution), when, on the same turn the following occurs:

1. China makes her bugged Declaration of War wherein I cannot choose the "You'll pay for this..." dialogue selection and a "China has declared war on you" dialogue window with a back button persists on the bottom of the screen.

2. Japan initiates a peace offering; but it's almost as if this second AI interaction during the turn processing somehow prevented the bug from fully taking into effect, because I don't experience any of the other symptoms such as the disappearing textures and the perpetually "loading" cursor, and I was able to save the game and reload, so the game was spared corruption somehow. (Possible clue as to what the issue is I hope?).

[Ramkhamhaeng_0020 BC-3200.zip](https://github.com/LoneGazebo/Community-Patch-DLL/files/2858004/Ramkhamhaeng_0020.BC-3200.zip)
[Ramkhamhaeng2.0 Logs.zip](https://github.com/LoneGazebo/Community-Patch-DLL/files/2858005/Ramkhamhaeng2.0.Logs.zip)

-------------------------

I noticed in the issue the other guy opened (#5160), he states he is using _Custom Advanced Setup_. I don't know if that's the same mod as _Really Advanced Setup_ or not.
